src/Geno/C++/GUI/Modals/NewItemModal.cpp

```diff
@@ -17,97 +17,114 @@
 
 #include "NewItemModal.h"
 
+#include "Auxiliary/ImGuiAux.h"
+#include "Auxiliary/STBAux.h"
 #include "GUI/Modals/OpenFileModal.h"
 
 #include <imgui.h>
 #include <imgui_internal.h>
 #include <misc/cpp/imgui_stdlib.h>
 
-enum RequestType
+NewItemModal::NewItemModal( void )
+	: m_IconFolder( STBAux::LoadImageTexture( "Icons/FolderColored.png" ) )
 {
-	None = 0,
-	RequestTypePath,
-	RequestTypeString,
-
-}; // RequestType
+	m_MinSize = ImVec2( 360.0f, 115.0f );
+	m_MaxSize = ImVec2( 530.0f, 115.0f );
+} // NewItemModal
 
 //////////////////////////////////////////////////////////////////////////
 
-void NewItemModal::RequestPath( std::string Title, std::filesystem::path DefaultLocation, void* pUser, PathCallback Callback )
+void NewItemModal::Show( const std::string Title, const char* pFilter, const std::filesystem::path& rLocation, Callback Callback )
 {
 	if( Open() )
 	{
-		m_Title       = std::move( Title );
-		m_Location    = std::move( DefaultLocation );
-		m_Callback    = reinterpret_cast< void* >( Callback );
-		m_pUser       = pUser;
-		m_RequestType = RequestTypePath;
-	}
+		m_Title    = Title;
+		m_pFilter  = pFilter;
```

Comment:
Looks kinda dangerous to store a copy of the char*. Do we even need to store it? It seems to only be used inside this function anyway

Reply:
i will change this